tests/python/pants_test/pantsd/test_pantsd_integration.py

```diff
@@ -22,19 +21,17 @@
 
 def launch_file_toucher(f):
   """Launch a loop to touch the given file, and return a function to call to stop and join it."""
-  executor = ThreadPoolExecutor(max_workers=1)
   halt = threading.Event()
-
   def file_toucher():
     while not halt.isSet():
       touch(f)
       time.sleep(1)
-
-  future = executor.submit(file_toucher)
+  thread = threading.Thread(target=file_toucher)
+  thread.start()
```

Comment:
Adding a `thread.daemon = True` before start would be good form to gaurd against abends.